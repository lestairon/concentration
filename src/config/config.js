import firebase from "firebase/app";
import "firebase/firestore";

const firebaseConfig = {
  apiKey: "AIzaSyAqp6h3M478xGTxf_fa5sqQHcuMjS0hqOc",
  authDomain: "concentration-64b0c.firebaseapp.com",
  databaseURL: "https://concentration-64b0c.firebaseio.com",
  projectId: "concentration-64b0c",
  storageBucket: "",
  messagingSenderId: "487946285632",
  appId: "1:487946285632:web:99ec7eef3a53e26a"
};

firebase.initializeApp(firebaseConfig);

export default firebase;
