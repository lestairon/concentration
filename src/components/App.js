import Board from "./Board";
import ScoreBoard from "./ScoreBoard";
import Timer from "./Timer";
import React from "react";
import { createStore, applyMiddleware, compose } from "redux";
import rootReducer from "../reducers";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { getFirestore, reduxFirestore } from "redux-firestore";
import config from "../config/config";

const enchancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  rootReducer,
  enchancers(
    applyMiddleware(thunk.withExtraArgument({ getFirestore })),
    reduxFirestore(config)
  )
);

const App = () => {
  return (
    <Provider store={store}>
      <Board numberOfCards={3} />
      <ScoreBoard />
      <Timer />
    </Provider>
  );
};
export default App;
