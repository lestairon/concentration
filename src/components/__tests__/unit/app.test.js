import App from "../../App";
import React from "react";
import { render, waitForDomChange, fireEvent } from "@testing-library/react";
import "jest-styled-components";
import { firestoreDB } from "../../../config/config";

jest.spyOn(firestoreDB, "collection").mockImplementation(() => ({
  onSnapshot: cb => {
    Promise.resolve(
      cb({
        docChanges: () => [
          {
            type: "added",
            doc: { data: () => ({ name: "test", score: 100 }) }
          }
        ]
      })
    );
  }
}));

test("renders correctly the Main component", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
test("renders popup before board", () => {
  const { getByLabelText, getByText, container } = render(<App />);
  const numberInput = getByLabelText(/number of cards/i);
  const submitButton = getByText(/Ok/i);
  fireEvent.change(numberInput, { target: { value: 3 } });
  fireEvent.click(submitButton);
  expect(container).toMatchSnapshot();
});
