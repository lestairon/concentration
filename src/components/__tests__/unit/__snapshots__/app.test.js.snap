// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the Main component 1`] = `
.c0 {
  background: #f3f3f3;
  border-radius: 10px;
  min-height: 60vh;
}

.c2 {
  --number-of-rows: 3;
  --number-of-columns: 4;
  display: grid;
  grid-template-rows: repeat( var(--number-of-rows), calc(85vmin / var(--number-of-rows)) );
  grid-template-columns: repeat( var(--number-of-columns), calc(85vmin / var(--number-of-columns)) );
  grid-row-gap: 2vmin;
  grid-column-gap: 2vmin;
  height: 100%;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  z-index: 2;
  pointer-events: none;
  position: absolute;
  font-family: Nunito;
  font-size: 0.98em;
  opacity: 0.67;
}

.c3 {
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  font-family: Nunito,sans-serif;
  text-align: center;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.5s;
  -webkit-transition: transform 0.5s;
  transition: transform 0.5s;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
}

.c3,
.c3 .cardback,
.c3 .cardfront {
  border-radius: 10px;
}

.c3:hover,
.c3 > *:hover {
  -webkit-transform: scale(1.015);
  -ms-transform: scale(1.015);
  transform: scale(1.015);
}

.c3.active {
  -webkit-transform: rotateY(180deg);
  -ms-transform: rotateY(180deg);
  transform: rotateY(180deg);
}

.c3 .cardback,
.c3 .cardfront {
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
}

.c3 .cardfront {
  position: absolute;
  top: 0;
  left: 0;
}

.c3 .cardback {
  -webkit-transform: rotateY(180deg);
  -ms-transform: rotateY(180deg);
  transform: rotateY(180deg);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  font-family: Nunito;
  position: fixed;
  bottom: 0;
}

@media (min-width:1025px) {
  .c0 {
    height: auto;
    width: 100%;
  }
}

@media (max-width:1920) {
  .c0 {
    height: 1080;
    width: 1920;
  }
}

<div>
  <div
    class="c0"
  >
    <h3
      class="c1"
    >
      Number of moves: 
      0
    </h3>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
      <div
        class="c3"
      >
        <img
          alt="Card"
          class="cardfront"
          draggable="false"
          src="education.svg"
        />
        <div
          class="cardback"
        />
      </div>
    </div>
    <div
      class="c4"
    >
      Seconds 0
    </div>
  </div>
</div>
`;

exports[`renders popup before board 1`] = `
.c0 {
  background: #f3f3f3;
  border-radius: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: inherit;
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  position: absolute;
  font-family: Nunito;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
  font-family: Nunito,sans-serif;
  background: #fff;
  border-radius: 10px;
  text-align: center;
  margin: 0.3em;
  width: 13vw;
  height: 40vh;
  min-width: 6.5em;
  max-height: 300px;
  min-height: 210px;
  cursor: pointer;
}

.c3.active {
  -webkit-animation: rotate-in-keyframes 0.6s;
  animation: rotate-in-keyframes 0.6s;
}

.c4 {
  font-family: Nunito;
  position: fixed;
  bottom: 0;
}

@media (min-width:1025px) {
  .c0 {
    height: auto;
    width: auto;
  }
}

@media (max-width:1920) {
  .c0 {
    height: 1080;
    width: 1920;
  }
}

<div>
  <div
    class="c0"
  >
    <h3
      class="c1"
    >
      0
    </h3>
    <div
      class="c2"
    >
      <div
        class="c3"
      />
      <div
        class="c3"
      />
      <div
        class="c3"
      />
      <div
        class="c3"
      />
      <div
        class="c3"
      />
      <div
        class="c3"
      />
    </div>
    <div
      class="c4"
    >
      Seconds 0
    </div>
  </div>
</div>
`;
